File: coco/src/main/java/coco/ep/m5/Divd.java
package coco.ep.m5;

import util.Node;
import util.Tree;

public interface Divd<FT> extends coco.ep.m4.Divd<FT>, Exp<FT> {

    default Tree astree() {
        return new Node(convert(this).id(), convert(getLeft()).astree(), convert(getRight()).astree());
    }

    default Integer id() {
        return 2130451;
    }
}

File: coco_csharp/CoCo/EP/M5/Divd.cs
using System.Collections.Generic;

namespace EP.M5 {
    public interface Divd<FT> : Exp<FT>, EP.M4.Divd<FT> {

        Tree Exp<FT>.astree() {
            return new Node(convert(this).id(), convert(getLeft()).astree(), convert(getRight()).astree());
        }

        int Exp<FT>.id() {
            return 2130451;
        }
    }
}